### PR TITLE
Remove space saver action + upgrade other actions

### PR DESCRIPTION
**Pull request by strickvl**
Overdue to upgrade / update some of our really ancient or deprecated actions. 

Also since we're on custom runners for ubuntu I removed the special action that was clearing space on the runner. (Should save a bit of time).

## Summary by CodeRabbit

- **Chores**
  - Updated GitHub Actions to newer versions across various workflow files for improved performance and security.

### CODE REVIEW

**Comment by strickvl** [comment]
@coderabbitai review

File: .github/actions/setup_environment/action.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ runs:
   using: "composite"
   steps:
     - name: Set up Python
-      uses: actions/setup-python@v4
+      uses: actions/setup-python@v5.0.0
```
Comment (avishniakov) [inline .github/actions/setup_environment/action.yml:32]: Here and in other places, why you are so specific as `5.0.0`? Why not just rely on majors like `5`?

**Review by avishniakov — APPROVED** [review]
One nitpick, rest good to go!

File: .github/actions/setup_environment/action.yml (hunk 1 above)
Comment (strickvl) [reply .github/actions/setup_environment/action.yml:32]: AFAIK you specify the literal tag name here and if there isn't a `v5` tag name it fails.

**Pull request merged**